api/src/paths/dwc/validate.ts

```diff
@@ -161,26 +77,78 @@ POST.apiDoc = {
   }
 };
 
-function prepDWCArchive(): RequestHandler {
+function getSubmissionS3Key(): RequestHandler {
   return async (req, res, next) => {
-    defaultLog.debug({ label: 'prepDWCArchive', message: 'files.length', files: req?.files?.length });
+    defaultLog.debug({ label: 'getSubmissionS3Key', message: 'params', files: req.body });
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    const occurrenceSubmissionId = req.body.occurrence_submission_id;
 
-    if (!req.files || !req.files.length) {
-      // no media objects included
-      throw new HTTP400('Missing upload data');
+    if (!occurrenceSubmissionId) {
+      throw new HTTP400('Missing required body param `occurrence_submission_id`.');
     }
 
-    if (req.files.length !== 1) {
-      // no media objects included
-      throw new HTTP400('Too many files uploaded, expected 1.');
+    try {
+      const sqlStatement = getSurveyOccurrenceSubmissionSQL(occurrenceSubmissionId);
+
+      if (!sqlStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
+
+      const response = await connection.query(sqlStatement.text, sqlStatement.values);
+
+      if (!response || !response.rowCount) {
+        throw new HTTP400('Failed to get survey occurrence submission');
+      }
+
+      const s3Key = response.rows[0].key;
+
+      req['s3Key'] = s3Key;
+
+      next();
+    } catch (error) {
+      defaultLog.debug({ label: 'getSubmissionS3Key', message: 'error', error });
+      throw error;
+    } finally {
+      connection.release();
     }
+  };
+}
+
+function getSubmissionFileFromS3(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'getSubmissionFileFromS3', message: 'params', files: req.body });
 
     try {
-      const rawMediaArray: Express.Multer.File[] = req.files as Express.Multer.File[];
+      const s3Key = req['s3Key'];
 
-      const rawMediaFile = rawMediaArray[0];
+      const s3File = await getFileFromS3(s3Key);
 
-      const mediaFiles = parseUnknownMedia(rawMediaFile);
+      if (!s3File) {
+        throw new HTTP400('Failed to get occurrence submission file');
```

Comment:
question: here is it a 400 cause its bad request from user or should it be 500 like we failed to get the file cause internal issue?

Reply:
Yeah I can update this.  

A bit of tech debt for the todo list is to do a general audit of all of our errors, and make sure they are informative. We can also tack on any additional error messaging as the second param in all of the HTTP errors.